src/lib/util/options.h

```diff
@@ -13,47 +13,62 @@
 
 #include "corefile.h"
 #include <unordered_map>
-
+#include <sstream>
 
 
 //**************************************************************************
 //  CONSTANTS
 //**************************************************************************
 
-// option types
-const uint32_t OPTION_TYPE_MASK       = 0x0007;       // up to 8 different types
-enum
-{
-	OPTION_INVALID,         // invalid
-	OPTION_HEADER,          // a header item
-	OPTION_COMMAND,         // a command
-	OPTION_BOOLEAN,         // boolean option
-	OPTION_INTEGER,         // integer option
-	OPTION_FLOAT,           // floating-point option
-	OPTION_STRING           // string option
-};
-
 // option priorities
 const int OPTION_PRIORITY_DEFAULT   = 0;            // defaults are at 0 priority
 const int OPTION_PRIORITY_LOW       = 50;           // low priority
 const int OPTION_PRIORITY_NORMAL    = 100;          // normal priority
 const int OPTION_PRIORITY_HIGH      = 150;          // high priority
 const int OPTION_PRIORITY_MAXIMUM   = 255;          // maximum priority
 
-const uint32_t OPTION_FLAG_INTERNAL = 0x40000000;
-
 
 //**************************************************************************
 //  TYPE DEFINITIONS
 //**************************************************************************
 
-// static structure describing a single option with its description and default value
-struct options_entry
+struct options_entry;
+
+// avoid obnoxious DirectSound issue
+#ifdef ERROR
+#undef ERROR
+#endif
```

Comment:
Is this really safe?  Would it be better to sidestep it by using abbreviated names like `WARN` and `ERR` instead?